src/test/unit_tests/jest_tests/test_list_object.test.js

```diff
@@ -0,0 +1,155 @@
+/* Copyright (C) 2016 NooBaa */
+/* eslint-disable no-undef */
+'use strict';
+
+
+const fs = require('fs');
+const path = require('path');
+const fs_utils = require('../../../util/fs_utils');
+const nb_native = require('../../../util/nb_native');
+const {TMP_PATH} = require('../../system_tests/test_utils');
+const { get_process_fs_context } = require('../../../util/native_fs_utils');
+
+const tmp_fs_path = path.join(TMP_PATH, 'test_list_object');
+const DEFAULT_FS_CONFIG = get_process_fs_context();
+
+// eslint-disable-next-line max-lines-per-function
+describe('manage list objct flow', () => {
+    describe('Telldir and Seekdir implementation', () => {
+        const list_dir_root = path.join(tmp_fs_path, 'list_dir_root');
+        const list_dir_1_1 = path.join(list_dir_root, 'list_dir_1_1');
+        const total_files = 4;
+
+        beforeAll(async () => {
+            await fs_utils.create_fresh_path(list_dir_root);
+            await fs_utils.create_fresh_path(list_dir_1_1);
+            for (let i = 0; i < total_files; i++) {
+                create_temp_file(list_dir_root, `test_${i}.json`, {test: test});
+            }
+        });
+
+        afterAll(async () => {
+            await fs_utils.folder_delete(`${list_dir_root}`);
+            await fs_utils.folder_delete(`${list_dir_1_1}`);
+        });
+
+        it('list dir files - telldir and seekdir.', async () => {
+            let tell_dir;
+            let dir_marker;
+            let total_dir_entries = 0;
+            let dir_entry;
+            let dir_handle = await nb_native().fs.opendir(DEFAULT_FS_CONFIG, list_dir_root);
+            // reak first read after 3 entries.
+            for (let i = 0; i <= 2; i++) {
+                dir_entry = await dir_handle.read(DEFAULT_FS_CONFIG);
+                if (!dir_entry) break;
+                tell_dir = await dir_handle.telldir(DEFAULT_FS_CONFIG);
```

Comment:
check here that the off returned in the direntry is the same as returned by telldir